app/src/main/java/com/kickstarter/features/search/viewmodel/SearchAndFilterViewModel.kt

```diff
@@ -91,16 +94,30 @@ class SearchAndFilterViewModel(
         this.errorAction = errorAction
     }
 
+    fun updateParamsToSearchWith(category: Category? = null, projectSort: DiscoveryParams.Sort = DiscoveryParams.Sort.POPULAR) {
+        val update = params.value.toBuilder()
+            .apply {
+                this.category(category)
+                this.sort(projectSort) // - Default sorting is popular
+            }
+            .build()
+
+        scope.launch {
+            _params.emit(update)
+        }
+    }
+
     /**
      * Update UIState with after executing Search query with latest params
      */
-    private suspend fun updateSearchResultsState() {
-        analyticEvents.trackSearchCTAButtonClicked(params.value)
+    private suspend fun updateSearchResultsState(params: DiscoveryParams) {
+        analyticEvents.trackSearchCTAButtonClicked(params)
```

Comment:
CTA analytic when tackling the pagination tickets will likely be moved to the init block or the public VM method to be launched only when the params are updated.